src/test/groovy/pl/tomaszdziurko/jvm_bloggers/blog_posts/domain/BlogPostSpec.groovy

```diff
@@ -0,0 +1,52 @@
+package pl.tomaszdziurko.jvm_bloggers.blog_posts.domain
+
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class BlogPostSpec extends Specification {
+
+    @Unroll
+    def "Should return \"#expectedState\" state when approved is #approved "(final Boolean approved, final String expectedState) {
+        given:
+            def blogPost = new BlogPost(approved: approved)
+        when:
+            def approvalState = blogPost.getApprovalState()
+        then:
+            approvalState == expectedState
+        where:
+            approved | expectedState
+            true     | "Approved"
+            false    | "Rejected"
+            null     | " -- "
+    }
+
+    @Unroll
+    def "Should return #expected when isModerated called for post with approved = #approved"(final Boolean approved, final boolean expected) {
+        given:
+            def blogPost = new BlogPost(approved: approved)
```

Comment:
Please avoid "def"s, for a consistenty we try to use stati types where possible.